src/tsdb.c

```diff
@@ -514,6 +514,7 @@ int SeriesCreateRulesFromGlobalConfig(RedisModuleCtx *ctx,
         if (RedisModule_KeyType(compactedKey) != REDISMODULE_KEYTYPE_EMPTY) {
             // TODO: should we break here? Is log enough?
             RM_LOG_WARNING(ctx, "Cannot create compacted key, key '%s' already exists", destKey);
+            RedisModule_FreeString(ctx, destKey);
```

Comment:
@danni-m  small leak 2

Reply:
@filipecosta90 im not sure this is a leak, the destKey is kept in inside the `CompactionRule` struct via `SeriesAddRule`

Reply:
@danni-m if you uncomment the fix line and run ``. You will receive the leak report as follows: `make test SIMPLE=1 VGD=1 TEST=test_globalconfigs.py`

```
353304:signal-handler (1608728599) Received SIGTERM scheduling shutdown...
353304:M 23 Dec 2020 13:03:19.868 # User requested shutdown...
353304:M 23 Dec 2020 13:03:19.875 # Redis is now ready to exit, bye bye...
==353304== 254 (80 direct, 174 indirect) bytes in 5 blocks are definitely lost in loss record 782 of 862
==353304==    at 0x483B7F3: malloc (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==353304==    by 0x152030: zmalloc (zmalloc.c:99)
==353304==    by 0x16415D: createObject (object.c:42)
==353304==    by 0x1DC6D4: RM_CreateStringPrintf (module.c:1050)
==353304==    by 0x5DCD520: SeriesCreateRulesFromGlobalConfig (tsdb.c:510)
==353304==    by 0x5DC5D9E: add (module.c:833)
==353304==    by 0x5DC606C: TSDB_add (module.c:876)
==353304==    by 0x1DB8C4: RedisModuleCommandDispatcher (module.c:643)
==353304==    by 0x1499BC: call (server.c:3354)
==353304==    by 0x14AC46: processCommand (server.c:3783)
==353304==    by 0x15DA71: processCommandAndResetClient (networking.c:1857)
==353304==    by 0x15DCDE: processInputBuffer (networking.c:1940)
```

Reply:
Looking at the code again you are right.